lib/libzfs/libzfs_util.c

```diff
@@ -596,27 +596,47 @@ zfs_strdup(libzfs_handle_t *hdl, const char *str)
  * Convert a number to an appropriately human-readable output.
  */
 void
-zfs_nicenum(uint64_t num, char *buf, size_t buflen)
+__zfs_nicenum(uint64_t num, char *buf, size_t buflen,
```

Comment:
I'm fine with renaming __zfs_nicenum() -> zfs_nicenum_impl() to follow convention.  After that you would have:

```
zfs_nicenum():  Wrapper function to niceprint base 1024
zfs_nicetime():  Wrapper function to niceprint time value
zfs_nicenum_impl(): Low-level function to niceprint values
```

However, I'm not clear on what `zfs_nicenum10` would do.  Is it for printing out the raw numerical values (i.e. the -p "parsable" option)?